Ryujinx.Graphics.Vulkan/DisposablePipeline.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Silk.NET.Vulkan;
+using System;
+
+namespace Ryujinx.Graphics.Vulkan
+{
+    struct DisposablePipeline : IDisposable
+    {
+        private readonly Vk _api;
+        private readonly Device _device;
+
+        public Pipeline Value { get; }
+
+        public DisposablePipeline(Vk api, Device device, Pipeline pipeline)
+        {
+            _api = api;
+            _device = device;
+            Value = pipeline;
+        }
+
+        public unsafe void Dispose()
+        {
+            _api.DestroyPipeline(_device, Value, null);
+        }
```

Comment:
This can be made "safe":
```suggestion
        public void Dispose()
        {
            _api.DestroyPipeline(_device, Value, Span<AllocationCallbacks>.Empty);
        }
```